wit-0.3.0-draft/system-clock.wit

```diff
@@ -1,27 +1,34 @@
 package wasi:clocks@0.3.0;
-/// WASI Wall Clock is a clock API intended to let users query the current
-/// time. The name "wall" makes an analogy to a "clock on the wall", which
-/// is not necessarily monotonic as it may be reset.
+/// WASI System Clock is a clock API intended to let users query the current
+/// time. The clock is not necessarily monotonic as it may be reset.
 ///
 /// It is intended to be portable at least between Unix-family platforms and
 /// Windows.
 ///
-/// A wall clock is a clock which measures the date and time according to
-/// some external reference.
+/// An "instant", or "exact time", is a point in time without regard to any time
+/// zone: just the time since a particular external reference point, often
+/// called an "epoch".
 ///
 /// External references may be reset, so this clock is not necessarily
 /// monotonic, making it unsuitable for measuring elapsed time.
 ///
 /// It is intended for reporting the current date and time for humans.
 @since(version = 0.3.0)
-interface wall-clock {
-    /// A time and date in seconds plus nanoseconds.
+interface system-clock {
+    /// An exact time in seconds plus nanoseconds.
     @since(version = 0.3.0)
-    record datetime {
+    record instant {
         seconds: u64,
```

Comment:
This is pre-existing, however while you're here, could you also change `seconds` from `u64` to `s64`? That better reflects the operating system APIs that it will be commonly implemented on, and supports timess from before 1970.

Reply:
Done. What about for durations? Is it intentional that durations cannot be negative? (See also https://github.com/tc39/proposal-temporal/issues/558.)

Reply:
Note that making durations potentially negative for the _monotonic_ clock is somewhat tricky unless `mark` is also changed to use `s64`, because the difference between the largest `u64` and `0` is not representable as an `s64` and hence a `s64` `duration` would not be able to represent all possible differences between two `mark`s.

(In practice this will never come up because no one is going to observe monotonic clock values which are 294 years apart, so I guess implementations could trap or something.)

Reply:
Right now, wasi-clocks has a much smaller surface area than `Temporal`, and it doesn't have any uses for negative durations. However, I'd be open to using a signed type. I think we could go either way. If it's signed, we just need to make sure we document that things like `resolution` don't return negative values, and document what `subscribe-durection` does given a negative value.

Reply:
Mostly I ask because [the readme](https://github.com/WebAssembly/wasi-clocks?tab=readme-ov-file#measuring-elapsed-time) has an example of a language creating a duration by subtracting two values from the clock, and it's not clear what the behavior would be if they'd reversed the order of those values. If the answer is "that's on the language to figure out" that's fine by me.

Reply:
Yes, the current design is that that's on the language to figure out.